.github/workflows/main.yaml

```diff
@@ -0,0 +1,36 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+
```

Comment:
We can only watch for pushes to main.  You can look at what's in the SD-Core charms Ex [here](https://github.com/canonical/sdcore-amf-operator/blob/main/.github/workflows/main.yaml#L7). Else we're running the workflow twice on PR's (because of both the pr and the push condition)

Reply:
I found it useful if we run the workflow without creating a PR which is in progress which is similar to https://github.com/canonical/sdcore-upf-operator/pull/39/files